Models/Library.cs

```diff
@@ -245,6 +246,33 @@ void exploreDataFiles()
 
         }
 
+        public List<string> samples => library.Keys.ToList();
+
+        public Measurement getSample(string name)
+        {
+            if (library.ContainsKey(name))
+                return library[name];
+            else 
+                return null;
+        }
+
+        public void addSampleToLibrary(string name, Measurement sample)
+        {
+            Measurement adjusted = sample;
+            if (sample.wavenumbers[0] == 400 && sample.wavenumbers.Length == 2008 && sample.wavenumbers.Last() == 2407)
```

Comment:
Not all XS models have the same spectral range.

Reply:
Noted and added to backlog